src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaActionPlanPersistenceAdapter.kt

```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.ActionPlanEntityMapper
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.ActionPlanRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.ActionPlan
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.ActionPlanNotFoundException
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.ActionPlanPersistenceAdapter
+
+@Component
+class JpaActionPlanPersistenceAdapter(
+  private val actionPlanRepository: ActionPlanRepository,
+  private val actionPlanMapper: ActionPlanEntityMapper,
+) : ActionPlanPersistenceAdapter {
+  override fun getActionPlan(prisonNumber: String): ActionPlan {
+    val entity = actionPlanRepository.findByPrisonNumber(prisonNumber)
+      ?: throw ActionPlanNotFoundException("No Action Plan found for prisoner [$prisonNumber]")
```

Comment:
I'm not sure about the persistence adapter throwing exceptions. This is what we have in the interface (I know it was my doing originally ... 🤔 )
```
/**
 * Persistence Adapter for [ActionPlan] instances.
 *
 * Implementations should use the underlying persistence of the application in question, eg: JPA, Mongo, Dynamo, Redis etc
 *
 * Implementations should not throw exceptions. These are not part of the interface and are not checked or handled by
 * [ActionPlanService].
 */
interface ActionPlanPersistenceAdapter {

  fun getActionPlan(prisonNumber: String): ActionPlan
}
```

I think the problem I was trying to prevent is that the persistence adapter impl (which is in the `app` codebase) is called by the service (which is in the `domain`) codebase.

Allowing the persistence adapter (app) to throw exceptions means that the service (domain) needs to handle them.

Perhaps this is OK in this case, where the exception is a `domain` exception ..... but I think I was trying to prevent persistence adapters throwing anything - especially exceptions pertaining to the underlying persistence technology ....

Reply:
I agree. See my other comment.